pkg/apis/distributed/v1alpha1/distributed_inference_service.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2025 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+)
+
+// DistributedInferenceServiceSpec is the top level for this resource.
+type DistributedInferenceServiceSpec struct {
+	// Replicas is the desired number of inferenceservice
+	// +kubebuilder:default=1
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
if we want to support disaggregating serving in this API, a top-level replicas might not be accurate to have as prefill and decode workers need to be scaled independently 

I think we need to brainstorm a little this new API for that use case